homeassistant/components/influx.py

```diff
@@ -0,0 +1,143 @@
+"""
+homeassistant.components.influx
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+InfluxDB component which allows you to send data to an Influx database.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/influx/
+
+Configuration:
+
+influx:
+  host: localhost
+  port: 8086
+  dbname: home_assistant
+  dbuser: DB_USER
+  dbuser_password: DB_USER_PASSWORD
+"""
+import logging
+import requests
+import socket
+
+import homeassistant.util as util
+from homeassistant.helpers import validate_config
+from homeassistant.const import (MATCH_ALL)
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "influx"
+DEPENDENCIES = ['recorder']
+
+INFLUX_CLIENT = None
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 8086
+DEFAULT_DATABASE = 'home_assistant'
+
+REQUIREMENTS = ['influxdb==2.10.0']
+
+CONF_HOST = 'host'
+CONF_PORT = 'port'
+CONF_DB_NAME = 'database'
+CONF_USERNAME = 'username'
+CONF_PASSWORD = 'password'
+
+
+def setup(hass, config):
+    """ Setup the Influx component. """
+
+    from influxdb import exceptions
+
+    if not validate_config(config, {DOMAIN: ['host']}, _LOGGER):
+        return False
+
+    conf = config[DOMAIN]
+
+    host = conf[CONF_HOST]
+    port = util.convert(conf.get(CONF_PORT), int, DEFAULT_PORT)
+    dbname = util.convert(conf.get(CONF_DB_NAME), str, DEFAULT_DATABASE)
+    username = util.convert(conf.get(CONF_USERNAME), str)
+    password = util.convert(conf.get(CONF_PASSWORD), str)
+
+    global INFLUX_CLIENT
+
+    try:
+        INFLUX_CLIENT = Influx(host, port, username, password, dbname)
+    except (socket.gaierror, requests.exceptions.ConnectionError):
+        _LOGGER.error("Database is not accessible. "
+                      "Please check your entries in the configuration file.")
+        return False
+
+    try:
```

Comment:
I think that creating the database is out of scope for HA. This should be done by the person setting up Influx.